SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestMultiServerUnrealComponents/SpatialTestMultiServerUnrealComponents.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestMultiServerUnrealComponents.h"
+
+#include "SpatialFunctionalTestFlowController.h"
+#include "SpatialGDKSettings.h"
+#include "TestUnrealComponents.h"
+#include "TestUnrealComponentsActor.h"
+
+#include "Kismet/GameplayStatics.h"
+#include "Net/UnrealNetwork.h"
+
+/**
+ * This test tests that static and dynamic component are properly resolved on expectant workers.
+ *
+ * The test includes 2 Server and 2 Clients.
+ * The flow is as follows:
+ * - Setup:
+ *  - The TestActor is spawned with both static and dynamic components.
+ *  - The components are initialized with unique starting values.
+ *  - The TestActor is possessed by Client1
+ * - Test:
+ *	- The owner client checks it can see the actor, components, and the values are correct.
+ *  - The non-auth server checks it can see the actor, components, and the values are correct.
+ *  - The non-owning client checks it can see the actor, components, and the values are correct.
+ * - Clean-up:
+ *	- The TestActor is destroyed.
+ */
+ASpatialTestMultiServerUnrealComponents::ASpatialTestMultiServerUnrealComponents()
+	: Super()
+{
+	Author = "Mike";
+	Description = TEXT("Test Unreal Component Replication in a MultiServer Context");
+}
+
+void ASpatialTestMultiServerUnrealComponents::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ThisClass, TestActor);
+}
+
+void ASpatialTestMultiServerUnrealComponents::PrepareTest()
+{
+	Super::PrepareTest();
+
+	bInitialOnlyEnabled = GetDefault<USpatialGDKSettings>()->bEnableInitialOnlyReplicationCondition;
+
+	// The Server spawns the TestActor and immediately after it creates and attaches the dynamic components
+	AddStep(TEXT("SpatialTestMultiServerUnrealComponents Spawn and Setup Test Actor"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		if (bInitialOnlyEnabled)
+		{
+			AddExpectedLogError(TEXT("Dynamic component using InitialOnly data. This data will not be sent."), 1, false);
+		}
+
+		AssertTrue(HasAuthority(), TEXT("Server requires auth"));
+		TestActor = GetWorld()->SpawnActor<ATestUnrealComponentsActor>(ActorSpawnPosition, FRotator::ZeroRotator, FActorSpawnParameters());
```

Comment:
We don't have an official documented position, but general convention has been to ignore said checks. I'll add them for this PR, but worth raising in the next katchup.

Reply:
Hm, for the special case of tests, I would say they are not that important, since we don't care that much if it crashes or errors, either way it's a severe failure.
But I guess just in general, we could have some convention, and follow it in our tests as well, just to keep things consistent.